### PR TITLE
Add and setup Jest testing framework

### PR DESCRIPTION
**Pull request by Johnnycus**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,5 +26,22 @@
   "bugs": {
     "url": "https://github.com/justin808/react-webpack-rails-tutorial/issues"
   },
-  "homepage": "https://github.com/justin808/react-webpack-rails-tutorial"
+  "homepage": "https://github.com/justin808/react-webpack-rails-tutorial",
+  "devDependencies": {
```
Comment (justin808) [inline package.json]: This should definitely go into client/package.json

File: app/assets/javascripts/components/__tests__/DemoComponent-test.jsx [hunk 2]
```diff
@@ -0,0 +1,6 @@
+jest.dontMock('../DemoComponent');
+
+describe('DemoComponent', function() {
+  it('should tell use it is a demo component', function() {
```
Comment (justin808) [inline app/assets/javascripts/components/__tests__/DemoComponent-test.jsx]: We want a test on a real component in the project.

File: app/assets/javascripts/components/DemoComponent.jsx [hunk 3]
```diff
@@ -0,0 +1,8 @@
+var DemoComponent = React.createClass({displayName: 'Demo Component',
```
Comment (justin808) [inline app/assets/javascripts/components/DemoComponent.jsx]: We don't need a demo component.

File: app/assets/javascripts/components/DemoComponent.jsx [hunk 4]
```diff
@@ -0,0 +1,8 @@
+var DemoComponent = React.createClass({displayName: 'Demo Component',
+  render: function() {
+    return <div>Demo Component</div>;
+  }
+});
+
+// each file will export exactly one component
+module.exports = DemoComponent;
```
Comment (justin808) [inline app/assets/javascripts/components/DemoComponent.jsx]: We're using es6
```
export default DemoComponent;
```

File: app/assets/javascripts/components/__tests__/DemoComponent-test.jsx [hunk 5]
```diff
@@ -0,0 +1,6 @@
+jest.dontMock('../DemoComponent');
```
Comment (justin808) [inline app/assets/javascripts/components/__tests__/DemoComponent-test.jsx]: Will Jest want to mock everything?

File: app/assets/javascripts/components/__tests__/preprocessor.js [hunk 6]
```diff
@@ -0,0 +1,7 @@
+var ReactTools = require('react-tools');
+
+module.exports = {
```
Comment (justin808) [inline app/assets/javascripts/components/__tests__/preprocessor.js]: Is this the best way to do this given that we're using babel and es6?

File: app/assets/javascripts/components/__tests__/DemoComponent-test.jsx (hunk 5 above)
Comment (Johnnycus) [inline app/assets/javascripts/components/__tests__/DemoComponent-test.jsx]: Yes, it's mock by default

File: app/assets/javascripts/components/__tests__/preprocessor.js (hunk 6 above)
Comment (Johnnycus) [inline app/assets/javascripts/components/__tests__/preprocessor.js]: It's what I found to transform JSX.

**Pull request closed without merging**